In [1]:
import pandas as pd
import numpy as np


In [2]:

data = pd.read_csv(r'analysed_df.csv')

df = pd.read_csv(r'KAG_energydata_complete.csv')


In [3]:
RH_out_column = df['RH_out']

# Add 'RH_out' column to data
data['RH_out'] = RH_out_column



In [4]:

data

Appliances         T1         T2         T3         T4         T5  \
0              60  19.890000  19.200000  19.790000  19.000000  17.166667   
1              60  19.890000  19.200000  19.790000  19.000000  17.166667   
2              50  19.890000  19.200000  19.790000  18.926667  17.166667   
3              50  19.890000  19.200000  19.790000  18.890000  17.166667   
4              60  19.890000  19.200000  19.790000  18.890000  17.200000   
...           ...        ...        ...        ...        ...        ...   
19730         100  25.566667  25.890000  27.200000  24.700000  23.200000   
19731          90  25.500000  25.754000  27.133333  24.700000  23.230000   
19732         270  25.500000  25.628571  27.050000  24.700000  23.230000   
19733         420  25.500000  25.414000  26.890000  24.700000  23.200000   
19734         430  25.500000  25.264286  26.823333  24.700000  23.200000   

              T6         T9      T_out  Tdewpoint  month  Total_Temperature  \
0       7.026667  17.033333   6.600000   5.300000      1          17.167407   
1       6.833333  17.066667   6.483333   5.200000      1          17.149630   
2       6.560000  17.000000   6.366667   5.100000      1          17.103704   
3       6.433333  17.000000   6.250000   5.000000      1          17.067037   
4       6.366667  17.000000   6.133333   4.900000      1          17.070741   
...          ...        ...        ...        ...    ...                ...   
19730  24.796667  23.200000  22.733333  13.333333      5          24.861481   
19731  24.196667  23.200000  22.600000  13.300000      5          24.774571   
19732  23.626667  23.200000  22.466667  13.266667      5          24.686138   
19733  22.433333  23.200000  22.333333  13.233333      5          24.499981   
19734  21.026667  23.200000  22.200000  13.200000      5          24.327810   

       PredictedEnergy  Cluster     RH_out  
0                 69.4        0  92.000000  
1                 64.9        0  92.000000  
2                 65.4        0  92.000000  
3                 68.5        0  92.000000  
4                 57.4        0  92.000000  
...                ...      ...        ...  
19730            117.6        4  55.666667  
19731            174.3        4  56.000000  
19732            238.7        1  56.333333  
19733            240.0        2  56.666667  
19734            323.6        2  57.000000  

[19735 rows x 15 columns]

In [5]:
# Group the data by the 'Cluster' column
grouped = data.groupby('Cluster')


In [7]:
# Create a new DataFrame with the desired columns and rows
profiled_data = pd.DataFrame({
    'min_Appliances': grouped['Appliances'].min(),
    'avg_Appliances': grouped['Appliances'].mean(),
    'max_Appliances': grouped['Appliances'].max(),
    'min_PredictedEnergy': grouped['PredictedEnergy'].min(),
    'avg_PredictedEnergy': grouped['PredictedEnergy'].mean(),
    'max_PredictedEnergy': grouped['PredictedEnergy'].max(),
    'min_T_out': grouped['T_out'].min(),
    'avg_T_out': grouped['T_out'].mean(),
    'max_T_out': grouped['T_out'].max(),
    'min_RH_out': grouped['RH_out'].min(),
    'avg_RH_out': grouped['RH_out'].mean(),
    'max_RH_out': grouped['RH_out'].max(),
    'min_Tdewpoint': grouped['Tdewpoint'].min(),
    'avg_Tdewpoint': grouped['Tdewpoint'].mean(),
    'max_Tdewpoint': grouped['Tdewpoint'].max()
}, index=pd.Series(range(grouped.ngroups), name='Clusters'))


In [8]:
# Transpose the DataFrame to make Cluster values as columns
profiled_data = profiled_data.T

In [9]:
# Print the new DataFrame
profiled_data

Clusters                      0           1           2            3  \
min_Appliances        10.000000  190.000000  340.000000   530.000000   
avg_Appliances        54.061737  258.613333  404.462445   647.304688   
max_Appliances        80.000000  330.000000  520.000000  1080.000000   
min_PredictedEnergy   17.300000   67.000000   68.900000    92.200000   
avg_PredictedEnergy   60.320074  238.410222  323.530633   485.556250   
max_PredictedEnergy  298.600000  499.600000  629.100000   893.200000   
min_T_out             -5.000000   -3.500000   -4.500000    -3.500000   
avg_T_out              6.845523    8.645931    7.767297     8.357487   
max_T_out             26.100000   23.200000   23.350000    23.233333   
min_RH_out            24.000000   30.000000   30.000000    31.833333   
avg_RH_out            82.107955   74.678568   76.519187    74.033854   
max_RH_out           100.000000  100.000000   99.500000    99.000000   
min_Tdewpoint         -6.600000   -5.800000   -5.800000    -5.700000   
avg_Tdewpoint          3.681596    3.970923    3.473740     3.579557   
max_Tdewpoint         15.500000   14.800000   14.800000    13.983333   

Clusters                      4  
min_Appliances        90.000000  
avg_Appliances       111.956313  
max_Appliances       180.000000  
min_PredictedEnergy   55.900000  
avg_PredictedEnergy  120.898392  
max_PredictedEnergy  527.400000  
min_T_out             -4.800000  
avg_T_out              8.798091  
max_T_out             25.400000  
min_RH_out            25.000000  
avg_RH_out            74.353230  
max_RH_out           100.000000  
min_Tdewpoint         -6.500000  
avg_Tdewpoint          4.018374  
max_Tdewpoint         15.200000

In [12]:
profiled_data.to_csv('profiling2.csv', index=True)

Cluster 0:

    -The appliances in this cluster have a relatively low power consumption (average Appliances of 54.06).
    -The outdoor temperatures (min_T_out, avg_T_out, max_T_out) are moderate, suggesting a temperate climate.
    -The outdoor humidity levels (min_RH_out, avg_RH_out, max_RH_out) are within a typical range.
    -The dewpoint temperatures (min_Tdewpoint, avg_Tdewpoint, max_Tdewpoint) are also moderate.
Recommendations: This cluster represents energy-efficient households, and their practices could be studied and promoted as best practices for energy conservation.

                                                                                                                                         
Cluster 1:

    -The appliances in this cluster have a significantly higher power consumption (average Appliances of 258.61).
    -The outdoor temperatures and humidity levels are slightly higher than in Cluster 0, suggesting warmer weather conditions.
Recommendations: Focus on improving the energy efficiency of appliances, particularly high-consumption ones. Consider implementing temperature control measures, such as better insulation or energy-efficient cooling systems.


Cluster 2:

    -This cluster represents households with very high energy consumption (average PredictedEnergy of 323.53).
    -The appliances in this cluster have the highest power consumption (average Appliances of 404.46).
    -The outdoor temperatures and humidity levels are similar to Cluster 1, indicating warmer weather conditions.
Recommendations: Prioritize the replacement or upgrade of energy-intensive appliances with more efficient models. Implement temperature control
measures, improve insulation, and consider renewable energy sources to reduce the overall energy consumption.


Cluster 3:

    -This cluster represents households with extremely high energy consumption (average PredictedEnergy of 485.56).
    -The appliances in this cluster have the second-highest power consumption (average Appliances of 647.30).
    -The outdoor temperatures and humidity levels are similar to Clusters 1 and 2, suggesting warmer weather conditions.
Recommendations: Conduct a comprehensive energy audit to identify the major contributors to high energy consumption. Consider implementing energy 
management systems, behavior modification programs, and possibly exploring alternative energy sources or on-site generation.



Cluster 4:

    -This cluster represents households with moderate energy consumption (average PredictedEnergy of 120.9).
    -The appliances in this cluster have a moderate power consumption (average Appliances of 111.96).
    -The outdoor temperatures and humidity levels are higher than in Cluster 0, indicating warmer weather conditions.
Recommendations: Explore opportunities for further energy savings through efficient appliance upgrades, temperature control measures, and behavioral changes related to energy usage.


The  final observation suggest that targeting high-consumption households (Clusters 1, 2, and 3) with tailored energy-saving measures, such as appliance upgrades, temperature control, insulation improvements, and renewable energy sources, could significantly reduce overall energy consumption. Additionally, promoting best practices from low-consumption households (Cluster 0) and implementing gradual improvements for moderate-consumption households (Cluster 4) can contribute to a comprehensive energy conservation strategy.